### ISSUE TITLE
Tracking implicit dependencies

### ISSUE THREAD

**Issue by jlstevens** [holoviz/holoviews#798]
Originally HoloViews was only targeting matplotlib and as a result we were happy to depend on a few things that matplotlib depended on. Likewise, IPython is now Jupyter and has split into a number of repositories.

Here are some dependencies we need to worry about for people using bokeh without matplotlib:
- `pyparsing` is need for the `%%opts` magic (included if matplotlib is installed).
- Annoyingly you now need `conda install nbformat` for the `NotebookArchive` (used to be included).
- `jinja2` (although this is still a dependency of ``notebook``) so only needed if you want to run unit tests without notebook installed.

I think that is all of them.

**Edit:** I didn't look carefully enough - it seems `nbformat` is included with `notebook`. So I suppose the key point is that `notebook` is required to run the unit tests.

**Comment by jbednar** [comment]
So, do we need to have two conda packages for HoloViews, one relying only on Param (and numpy?), and the other building on that to add these non-core dependencies?

**Comment by jlstevens** [comment]
> So, do we need to have two conda packages for HoloViews, one relying only on Param (and numpy?), and the other building on that to add these non-core dependencies?

I suppose so. I just wish conda supported something like ``extras_require``!

**Comment by jbednar** [comment]
Agreed.  I was just talking to the conda people about noarch support, another glaring problem, and it's clear that they are overloaded right now, so I wouldn't expect it to get better soon!
